From 14e50fbbf6a83daa0f78844914b684ad29f9ecd6 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 31 Jan 2024 15:49:00 +0100
Subject: [PATCH] notification: Fix libnotify warning in Flatpak

When blocking direct access to the notification service, and relying on
libnotify 0.8's implicit portal support, we got those warnings:

(rhythmbox:2): libnotify-WARNING **: 15:02:43.424: Running in confined mode, using Portal notifications. Some features and hints won't be supported
libnotify-Message: 15:02:43.424: Category is not available when using Portal Notifications
libnotify-Message: 15:02:43.425: Category is not available when using Portal Notifications
libnotify-Message: 15:02:43.454: Category is not available when using Portal Notifications

We should only set the category when not using Flatpak.

See https://github.com/flathub/org.gnome.Rhythmbox3/pull/65
---
 plugins/notification/rb-notification-plugin.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/plugins/notification/rb-notification-plugin.c b/plugins/notification/rb-notification-plugin.c
index 40c62a8ac827..f2bc3ce513a7 100644
--- a/plugins/notification/rb-notification-plugin.c
+++ b/plugins/notification/rb-notification-plugin.c
@@ -59,6 +59,7 @@ struct _RBNotificationPlugin
 
 	gchar *notify_art_path;
 	RBExtDBKey *notify_art_key;
+	gboolean is_flatpak;
 	NotifyNotification *notification;
 	NotifyNotification *misc_notification;
 	gboolean notify_supports_actions;
@@ -199,7 +200,7 @@ do_notify (RBNotificationPlugin *plugin,
 					      g_variant_new_string (image_uri));
 	}
 
-        if (playback)
+        if (!plugin->is_flatpak && playback)
         	notify_notification_set_category (notification, "x-gnome.music");
         notify_notification_set_hint (notification, "desktop-entry",
                                       g_variant_new_string ("org.gnome.Rhythmbox3"));
@@ -649,8 +650,12 @@ impl_deactivate	(PeasActivatable *bplugin)
 }
 
 static void
-rb_notification_plugin_init (RBNotificationPlugin *plugin)
+rb_notification_plugin_init (RBNotificationPlugin *bplugin)
 {
+	RBNotificationPlugin *plugin;
+
+	plugin = RB_NOTIFICATION_PLUGIN (bplugin);
+	plugin->is_flatpak = g_file_test ("/.flatpak-info", G_FILE_TEST_EXISTS);
 }
 
 G_MODULE_EXPORT void
-- 
2.43.0

